File: leetcode/two-pointers/py/724.find-pivot-index.py
#
# @lc app=leetcode id=724 lang=python3
#
# [724] Find Pivot Index
#

# @lc code=start
class Solution:
	def pivotIndex(self, nums: List[int]) -> int:
		left, right = 0, 0
		for num in nums[1:]:
			right += num
		for i in range(lenw(nums)):
			if left == right:
				return i
			left += nums[i]
			if i == len(nums) - 1:
				break ;
			right -= nums[i + 1]
		return -1
# @lc code=end

File: leetcode/two-pointers/cpp/724.find-pivot-index.cpp
/*
 * @lc app=leetcode id=724 lang=cpp
 *
 * [724] Find Pivot Index
 */

// @lc code=start
class Solution {
public:
	int pivotIndex(vector<int>& nums) {
		int	left = 0;
		int right = 0;
		for (auto n : nums | views::drop(1)) {
			right += n;
		}
		for (auto it = nums.begin(); it != nums.end(); it++) {
			if (left == right)
				return (it - nums.begin());
			left += *it;
			right = (it + 1 == nums.end() ? 0 : right - *(it + 1));
		}
		return -1;
	}
};
// @lc code=end
